### DISCUSSION TITLE
can i use personal_sign with rainbowkit?

### DISCUSSION THREAD

**Discussion by blackrabbit944** [rainbow-me/rainbowkit#473] [Q&A]
i wanna use personal_sign to verify user own wallet account, is there any way to use personal_sign with rainbowkit?

**Comment by jxom** [comment]
Yes, you can use the `useSignMessage` hook from `wagmi`.

https://wagmi.sh/docs/hooks/useSignMessage

**Reply by blackrabbit944** [reply]
Thanks for the reply.
But I still have a question, if I want the signature to be called immediately after the user clicks on connect wallet and selects metamask in the popup box, otherwise it will not show that the wallet address is connected, is this possible?

That is, within the rainbowkit component, the personal_sign method is invoked when the metamask is clicked
